src/mol-model/structure/structure/unit/bonds/inter-compute.ts

```diff
@@ -82,11 +82,22 @@ function findPairBonds(unitA: Unit.Atomic, unitB: Unit.Atomic, props: BondComput
 
                 const _bI = SortedArray.indexOf(unitB.elements, bI) as StructureElement.UnitIndex;
                 if (_bI < 0) continue;
-                if (type_symbolA.value(aI) === 'H' && type_symbolB.value(bI) === 'H') continue;
+
+                const aeI = getElementIdx(type_symbolA.value(aI));
+                const beI = getElementIdx(type_symbolA.value(bI));
+                // only element-based test when maxDistance !== -1
+                if (maxDistance !== -1 && isHydrogen(aeI) && isHydrogen(beI)) continue;
 
                 const d = distance[i];
                 const dist = getDistance(unitA, aI, unitB, bI);
-                if ((d !== -1 && equalEps(dist, d, 0.5)) || dist < maxDistance) {
+                const pairingThreshold = getPairingThreshold(
+                    aeI, beI, getElementThreshold(aeI), getElementThreshold(beI)
+                );
+
+                if ((d !== -1 && equalEps(dist, d, 0.5)) ||
+                    (maxDistance !== -1 && dist < maxDistance) ||
+                    dist < pairingThreshold
+                ) {
```

Comment:
The logic here seems a bit fishy. I think it should be something like this instead:

```ts
if (isHydrogen(aeI) && isHydrogen(beI)) {
  // TODO: check if molecule has only 2 bonds and both are hydrogen to take care of edge case?
  continue;
}

if (maxDistance < 0) {
  // pairingThreshold based check
} else if (dist < maxDistance) {
  builder.add(_aI, _bI, { order: order[i], flag: flag[i] });
}
```

Reply:
yes, thanks, gave it an overhaul